### PR TITLE
Fixed PVC creation and GCP issues

### PR DESCRIPTION
**Pull request by bipuladh**
- Added missing field for GCP target bucket
- Added missing help text for GCP upload
- Fixes payload for GCP and S3 Endpoint Type

### CODE REVIEW

**Comment by gnehapk** [comment]
Please add bugzilla id to this PR

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx [hunk 1]
```diff
@@ -538,20 +554,40 @@ const CreateBackingStoreForm: React.FC<CreateBackingStoreFormProps> = withHandle
       },
       spec: {
         type: typeNoobaaMap[provider],
+        ssl: false,
+      },
+    };
+    if (provider === 'PVC') {
+      // eslint-disable-next-line
+      bsPayload.spec['pvPool'] = {
```
Comment (gnehapk) [inline frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: why not `bsPayload.spec.pvPool`? Then you don't need disable eslint for next line.

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx [hunk 2]
```diff
@@ -538,20 +554,40 @@ const CreateBackingStoreForm: React.FC<CreateBackingStoreFormProps> = withHandle
...
         },
-        ssl: false,
-      },
-    };
-
+      };
+    }
+    if (provider === 'S3 Compatible') {
+      // eslint-disable-next-line
+      bsPayload.spec['s3Compatible'] = {
```
Comment (gnehapk) [inline frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: Same as above comment

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx [hunk 3]
```diff
@@ -538,20 +554,40 @@ const CreateBackingStoreForm: React.FC<CreateBackingStoreFormProps> = withHandle
...
             namespace,
           },
         },
-        ssl: false,
-      },
-    };
-
+      };
+    }
+    if (provider === 'S3 Compatible') {
```
Comment (gnehapk) [inline frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: Same as above comment

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx [hunk 4]
```diff
@@ -538,20 +554,40 @@ const CreateBackingStoreForm: React.FC<CreateBackingStoreFormProps> = withHandle
...
+      // eslint-disable-next-line
+      bsPayload.spec['s3Compatible'] = {
+        // eslint-disable-next-line
+        ...bsPayload.spec['s3Compatible'],
```
Comment (gnehapk) [inline frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: Same as above comment

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx (hunk 3 above)
Comment (gnehapk) [inline frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: Move all these strings to constants and the use here.  Same applies to rest of the code.

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx (hunk 1 above)
Comment (bipuladh) [reply frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: it dont work like that. Have a look  at spec object.

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx (hunk 2 above)
Comment (bipuladh) [reply frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: same as above

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx (hunk 4 above)
Comment (bipuladh) [reply frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: same as above

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx [hunk 5]
```diff
@@ -538,20 +554,40 @@ const CreateBackingStoreForm: React.FC<CreateBackingStoreFormProps> = withHandle
...
+        endpoint: providerDataState.endpoint,
+      };
+    }
+    // Add region in the end
     if (provider === 'AWS S3') {
```
Comment (gnehapk) [inline frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: Move it to constant

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx (hunk 3 above)
Comment (bipuladh) [reply frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: ack

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx (hunk 5 above)
Comment (bipuladh) [reply frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: ack

File: frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx (hunk 1 above)
Comment (gnehapk) [reply frontend/packages/noobaa-storage-plugin/src/components/create-backingstore-page/create-bs.tsx]: ack

**Pull request merged**